ztp/siteconfig-generator/siteConfig/siteConfig_test.go

```diff
@@ -370,3 +371,135 @@ spec:
 		}
 	}
 }
+
+func TestPartitions_UnmarshalYAML(t *testing.T) {
+	type fields struct {
+		MountPoint    string
+		Size          int
+		Start         int
+		MountFileName string
+		Label         string
+	}
+	type args struct {
+		unmarshal func(interface{}) error
+	}
+
+	tests := []struct {
+		name           string
+		fields         fields
+		args           args
+		wantErr        bool
+		expectedResult *Partitions
+		expectedError  string
+	}{
+		{
+			name: "check if the mount file name is generated correctly",
+			args: args{unmarshal: func(i interface{}) error {
```

Comment:
It might be more robust to follow the code path that the yaml takes, so instead of mocking out the `unmarshal` function it would help catch more problems if we use yaml string, exactly what the user would define, and follow that code path, what do you think?

Reply:
Updated the tests like we discussed